src/controller/python/chip/clusters/ClusterObjects.py

```diff
@@ -275,3 +275,59 @@ def _cluster_object(cls) -> ClusterObject:
                                   )
                               ],
                               bases=(ClusterObject,))
+
+
+class ClusterEventDescriptor:
+    '''
+    The ClusterEventDescriptor is used for holding an event's metadata like its cluster id, event id and its type.
+
+    Users should not initialize an object based on this class. Instead, users should pass the subclass objects to tell some methods what they want.
+
+    The implementation of this functions is quite tricky, it will create a cluster object on-the-fly, and use it for actual encode / decode routine to save lines of code.
+    '''
+    @classmethod
+    def ToTLV(cls, tag: Union[int, None], value):
+        writer = tlv.TLVWriter()
+        wrapped_value = cls._cluster_object(Value=value)
+        cls.event_type.PutFieldToTLV(tag,
+                                     asdict(wrapped_value)['Value'], writer, '')
+        return writer.encoding
+
+    @classmethod
+    def FromTLV(cls, tlvBuffer: bytes):
+        obj_class = cls._cluster_object
+        return obj_class.FromDict(obj_class.descriptor.TagDictToLabelDict('', {0: tlv.TLVReader(tlvBuffer).get().get('Any', {})})).Value
+
+    @classmethod
+    def FromTagDictOrRawValue(cls, val: Any):
+        obj_class = cls._cluster_object
+        return obj_class.FromDict(obj_class.descriptor.TagDictToLabelDict('', {0: val})).Value
+
+    @ChipUtility.classproperty
+    def cluster_id(self) -> int:
+        raise NotImplementedError()
+
+    @ChipUtility.classproperty
+    def event_id(self) -> int:
+        raise NotImplementedError()
+
+    @ChipUtility.classproperty
+    def event_type(cls) -> ClusterObjectFieldDescriptor:
+        raise NotImplementedError()
+
+    @ChipUtility.classproperty
+    def _cluster_object(cls) -> ClusterObject:
```

Comment:
Why is this needed? It was needed for attributes since each attribute was of type `ClusterObjectFieldDescriptor`, but in this case, you have a perfectly valid `ClusterObjectDescriptor` available here...